### ISSUE TITLE
Stabilize JaCoCo usage

### ISSUE THREAD

**Issue by shanshin** [Kotlin/kotlinx-kover#11]
When using JaCoCo as an agent of the JVM, the plugin must support at least the following functionality:
- generation of XML report
- generation of HTML report
- filtering by classes and functions in the report
- support for multi-platform applications
- support for android applications
- code coverage verification rules
- any other plugin features added later

For plugin version `0.2.2`, the agent for use is specified in the Gradle configuration in the extension to the test task e.g. for `build.gradle.kts`
```
tasks.test {
    extensions.configure(kotlinx.kover.KoverTaskExtension::class) {
        useJacoco = true // use JaCoCo as coverage agent and report generator
    }
}
```

**Comment by JavierSegoviaCordoba** [comment]
I am trying to use Kover with sonarqube but the task is not generating the XML when Jacoco engine is selected. IntelliJ one is working tho. Is this XML compatible with cloud platforms that use jacoco one?

**Comment by JavierSegoviaCordoba** [comment]
I am generating it, I think the cache was the problem it was not being regenerated. There is an issue about Kover should be usable inside Codecov, if that is fixed, it will work on Sonarqube too, no?

**Comment by shanshin** [comment]
Implemented in versions > 0.5.x
